lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp

```diff
@@ -223,6 +223,58 @@ class RecomposeSplitTensorGetItemOp
     return success();
   }
 };
+
+class RecomposeChunkListUnpack : public OpRewritePattern<PrimListUnpackOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(PrimListUnpackOp op,
+                                PatternRewriter &rewriter) const override {
+    // recompose AtenChunkOp + PrimListUnpackOp to AtenSliceTensorOps
+    auto chunk = dyn_cast<AtenChunkOp>(op.getOperand().getDefiningOp());
+    if (!chunk)
+      return failure();
+    if (isListPotentiallyMutated(chunk.getResult()))
+      return failure();
```

Comment:
nit: can you use `rewriter.notifyMatchFailure(...)` with an error message? Makes it easier to debug from `-debug` logs